### PR TITLE
Fixes mypy errors

### PR DESCRIPTION
**Pull request by sverhoeven**
Except for ewatercycle/forcing/wflow.py, those will be fixed in #82

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,6 +58,7 @@
             'pytest-cov',
             'pytest-mypy',
             'pytest-runner',
+            'types-python-dateutil',
```
Comment (Peter9192) [inline setup.py:61]: Should we also mock this in docs/conf.py?

File: ewatercycle/util.py [hunk 2]
```diff
@@ -71,12 +72,12 @@ def lat_lon_boundingbox_to_variable_indices(model, variable, latMin, latMax, lon
...
-    latModel = model.get_grid_x(model.get_var_grid(variable))
-    lonModel = model.get_grid_y(model.get_var_grid(variable))
-    nx = len(latModel)
+    lat_model = model.get_grid_x(model.get_var_grid(variable))
+    lon_model = model.get_grid_y(model.get_var_grid(variable))
+    nx = len(lat_model)
 
-    idx = [i for i,v in enumerate(latModel) if ((v > latMin) and (v < latMax))]
-    idy = [i for i,v in enumerate(lonModel) if ((v > lonMin) and (v < lonMax))]
+    idx = [i for i,v in enumerate(lat_model) if ((v > latMin) and (v < latMax))]
```
Comment (bvreede) [inline ewatercycle/util.py]:
```suggestion
    idx = [i for i,v in enumerate(lat_model) if ((v > lat_min) and (v < lat_max))]
```

Comment (bvreede) [reply ewatercycle/util.py]: should these variable names not also be adjusted?

File: ewatercycle/util.py [hunk 3]
```diff
@@ -71,12 +72,12 @@ def lat_lon_boundingbox_to_variable_indices(model, variable, latMin, latMax, lon
...
+    idy = [i for i,v in enumerate(lon_model) if ((v > lonMin) and (v < lonMax))]
```
Comment (bvreede) [inline ewatercycle/util.py]:
```suggestion
    idy = [i for i,v in enumerate(lon_model) if ((v > lon_min) and (v < lon_max))]
```

**Review by bvreede — COMMENTED** [review]
FWIW: I checked this branch locally and did not find further mypy errors (except in `wflow.py`).

File: setup.py (hunk 1 above)
Comment (sverhoeven) [reply setup.py:61]: Not needed, build https://ewatercycle--102.org.readthedocs.build/en/102/ succeeded. Also we dont import that package, its something mypy needed.

File: ewatercycle/util.py (hunk 2 above)
Comment (sverhoeven) [reply ewatercycle/util.py]: Hmm, pycharm did not complain about those, but yes let's also adjust those vars.

**Pull request merged**